scene/2d/gpu_particles_2d.cpp

```diff
@@ -781,6 +797,59 @@ void GPUParticles2D::_notification(int p_what) {
 	}
 }
 
+#ifdef TOOLS_ENABLED
+void GPUParticles2D::_draw_emission_gizmo() {
+	Ref<ParticleProcessMaterial> pm = process_material;
+	if (pm.is_null()) {
+		return;
+	}
+	draw_set_transform(
+			Vector2(pm->get_emission_shape_offset().x, pm->get_emission_shape_offset().y),
+			0.0,
+			Vector2(pm->get_emission_shape_scale().x, pm->get_emission_shape_scale().y));
+
+	switch (pm->get_emission_shape()) {
+		case ParticleProcessMaterial::EmissionShape::EMISSION_SHAPE_BOX: {
+			Vector2 extends2d = Vector2(pm->get_emission_box_extents().x, pm->get_emission_box_extents().y);
+			draw_rect(Rect2(extends2d, extends2d * 2.0), Color(0.8, 0.7, 0.4, 0.4), false);
```

Comment:
Could the color be saved somewhere in a variable since it seems to be used a lot in several places?

Reply:
Good point. Done :).